samples/subsys/fs/fat_fs/prj_stm32l496g_disco.conf

```diff
@@ -0,0 +1,5 @@
+CONFIG_DISK_ACCESS=y
+CONFIG_LOG=y
+CONFIG_FILE_SYSTEM=y
+CONFIG_FAT_FILESYSTEM_ELM=y
+CONFIG_PRINTK=y
```

Comment:
IIC These are all already available in prj.conf, so this file should'nt be needed

Reply:
This conf file isn't under `boards/`, so it gets used instead of `prj.conf`. I proposed a refactoring of this sample's conf files in #24787

Reply:
This seems cleaner indeed, txs @MaureenHelm.
@anthonybrandon can you use https://github.com/zephyrproject-rtos/zephyr/pull/24787/commits/3649c16046c083e2f6d3951791cc4916180901d4 and do you changes on top?